File: typescript-client-sdk/src/sdk/models/shared/connection.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum ConnectionIntegrationIdEnum {
    Salesforce = "salesforce",
    Hubspot = "hubspot",
    Pipedrive = "pipedrive"
}

export enum ConnectionStatusEnum {
    NewConnection = "NEW_CONNECTION",
    InitialSync = "INITIAL_SYNC",
    Ready = "READY"
}


export class Connection extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=connectionId" })
  connectionId?: string;

  @SpeakeasyMetadata({ data: "json, name=createdTime" })
  createdTime?: string;

  @SpeakeasyMetadata({ data: "json, name=integrationId" })
  integrationId?: ConnectionIntegrationIdEnum;

  @SpeakeasyMetadata({ data: "json, name=lastActivityDate" })
  lastActivityDate?: string;

  @SpeakeasyMetadata({ data: "json, name=nativeOrgURL" })
  nativeOrgURL?: string;

  @SpeakeasyMetadata({ data: "json, name=status" })
  status?: ConnectionStatusEnum;
}

File: python-client-sdk/src/orbapi/models/shared/connection.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from orbapi import utils

class ConnectionIntegrationIDEnum(str, Enum):
    SALESFORCE = "salesforce"
    HUBSPOT = "hubspot"
    PIPEDRIVE = "pipedrive"

class ConnectionStatusEnum(str, Enum):
    NEW_CONNECTION = "NEW_CONNECTION"
    INITIAL_SYNC = "INITIAL_SYNC"
    READY = "READY"


@dataclass_json
@dataclasses.dataclass
class Connection:
    connection_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('connectionId') }})
    created_time: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('createdTime') }})
    integration_id: Optional[ConnectionIntegrationIDEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('integrationId') }})
    last_activity_date: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('lastActivityDate') }})
    native_org_url: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('nativeOrgURL') }})
    status: Optional[ConnectionStatusEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    
